File: Python/dsaOne/Arrays_and_Hashing/groupAnagrams.py
#https://leetcode.com/problems/group-anagrams/description/

def groupAnagrams(strs):
    hm = {}
    for i in strs:
        res = ''.join(sorted(i))

        if res in hm:
            hm[res].append(i)
        else:
            hm[res] = [i]
        
    res = []
    for j in hm:
        res.append(hm[j])
    
    return res

print(groupAnagrams(['ate', 'eat', 'tan', 'can', 'ant']))
print(groupAnagrams([]))
        
    

File: Python/dsaOne/Javascript/Arrays_and_Hashing/groupAnagrams.js
const groupAnagrams = (strings) => {
    let hm = new Map()
    let res = []

    for(let i = 0; i < strings.length; i++) {
        let key = strings[i].split('').sort().join('')
        
        if(hm.has(key)) {
            let temp = hm.get(key)
            temp.push(strings[i])
            hm.set(key, temp)
        }
        else {
            hm.set(key, [strings[i]])
        }
    }

    hm.forEach(function (v,k) {
        res.push(v)
    })

    return res

}


console.log(groupAnagrams(["eat","tea","tan","ate","nat","bat"]))
console.log(groupAnagrams([""]))
console.log(groupAnagrams(["a"]))